docs/reference/snapshot-restore/apis/put-repo-api.asciidoc

```diff
@@ -176,11 +176,8 @@ in snapshots. Data files are not compressed. Defaults to `true`.
 
 `max_number_of_snapshots`::
 (Optional, integer)
-Maximum number of snapshots the repository can contain. Defaults to `500`.
-+
-WARNING: We do not recommend increasing `max_number_of_snapshots`. Larger
-snapshot repositories may degrade master node performance and cause stability
-issues. Instead, delete older snapshots or use multiple repositories.
+Maximum number of snapshots the repository can contain.
```

Comment:
I guess an alternative could be dropping the setting altogether again, assuming that we'll be infinitely scalable eventually. But I figured this provided a somewhat reasonable fail-safe for now that people can use to guard against problematic SLM policies so I figured we could keep it.